usr/src/man/man1m/zpool.1m

```diff
@@ -1342,6 +1382,72 @@ Stop scrubbing.
 .El
 .It Xo
 .Nm
+.Cm trim
+.Op Fl r Ar rate | Fl s
+.Ar pool Ns ...
+.Xc
+Initiates a manual TRIM operation on all of the free space of a pool.
+This informs the underlying storage devices of all of the blocks that
+the pool no longer considers allocated, thus allowing thinly provisioned
+storage devices to reclaim them. Please note that this collects all
+space marked as "freed" in the pool immediately and doesn't wait the
+.Nm zfs_txgs_per_trim
+delay as automatic TRIM does. Hence, this can limit pool corruption recovery
+options during and immediately following the manual TRIM to 1-2 TXGs into the
+past (instead of the standard 32-64 of automatic TRIM). This approach,
+however, allows you to recover the maximum amount of free space from the pool
+immediately without having to wait.
+.Pp
+Also note that a manual TRIM operation can be initiated irrespective of
+the
+.Sy autotrim
+zpool property setting. It does, however, respect the
+.Sy forcetrim
+zpool property.
+.Pp
+A manual TRIM operation does not conflict with an ongoing scrub, but it can
+put significant I/O stress on the underlying vdevs. A resilver, however,
+automatically stops a manual TRIM operation. You can manually reinitiate the
+TRIM operation after the resilver has started, by simply reissuing the
+.Nm zpool Cm trim
+command.
+.Pp
+Adding a vdev during TRIM is supported, although the progression display in
+.Nm zpool Cm status
+might not be entirely accurate in that case (TRIM will complete before
+reaching 100%). Removing or detaching a vdev will prematurely terminate
+a manual TRIM operation.
+.Bl -tag -width Ds
+.It Fl r Ar rate
+Controls the speed at which the TRIM operation progresses. Without this
+option, TRIM is executed in parallel on all top-level vdevs as quickly
+as possible. This option allows you to control how fast (in bytes per
+second) the TRIM is executed. This rate is applied on a per-vdev basis,
+i.e. every top-level vdev in the pool tries to match this speed. The rate
+argument supports common multiplier suffixes such as `k' for kilobytes,
+`m' for megabytes and `g' for gigabytes per second.
+.Pp
+Due to limitations in how the algorithm is designed, TRIMs are executed
+in whole-metaslab increments. Each top-level vdev contains approximately
+200 metaslabs, so a rate-limited TRIM progresses in steps. It TRIMs
+one metaslab completely and then waits for a while so that over the
+whole device, the speed averages out.
```

Comment:
I'm gonna kill it, since the it no longer applies and I just forgot to remove this section. It described an older implementation of the algorithm which worked in metaslab increments. I've since done some refinements there so that the granularity is much more fine (128MB increments, actually).